From 6af63eda4d4ce73cef6273b875471a2ac7ae95fc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 09 Nov 2020 13:30:12 +0200
Subject: [PATCH] [SW-12755] gaudi: security: unprotect secured bits

Unprotect the IF RTR CTRL secured bits explicitly.
This is in preparation for dynamic security and specifically for the IF
RTR CTRL secured bits protection which we will toggle during the dynamic
protection.
This is the second set of bits that was missed.

Change-Id: I23264c5b5b05bfe39b41ea41f3a08ee4606c44de
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 2d2be5f..a752def 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -656,9 +656,11 @@
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x880 - 0x900 allowed access */
+	prot_base += SNGL_REG_PROT_SIZE;
+	regmap_writel(~0, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x900 - 0x980 as following: */
-	prot_base += SNGL_REG_PROT_SIZE * 2;
+	prot_base += SNGL_REG_PROT_SIZE;
 	mask = ~(PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_7) &
 		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_8) &
 		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_SEC_MASK_HIGH_AR_9) &
